apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/java/file/OperationBuilder.kt

```diff
@@ -244,5 +247,39 @@ internal class OperationBuilder(
         context.resolver.resolveOperationSelections(operation.name)
     )
   }
-}
 
+  private fun TypeSpec.Builder.addBuilder(context: JavaContext): TypeSpec.Builder {
+    addMethod(Builder.builderFactoryMethod())
+
+    val operationClassName = ClassName.get("", simpleName)
+
+    if (operation.variables.isEmpty()) {
+      return addType(
+        Builder(
+          targetObjectClassName = operationClassName,
+          fields = emptyList(),
+          fieldDefaultValues = emptyMap(),
+          fieldJavaDocs = emptyMap(),
+          context = context
+        ).build()
+      )
+    }
+
+    operation.variables
+      .map {
+        context.layout.propertyName(it.name.decapitalizeFirstLetter()) to context.resolver.resolveIrType(it.type)
```

Comment:
GraphQL is case sensitive so we need to respect the variable case to avoid clashes:
```suggestion
        context.layout.propertyName(it.name) to context.resolver.resolveIrType(it.type)
```

```graphql
query GetHero($id: String!, $ID: String!) {
  hero1: hero(id: $id) { name }
  hero2: hero(id: $ID) { name }
}
```

This is a silly example and unlikely to happen in variables but this is happening for real on field names and enum values so it's more consistent to use the GraphQL case for this.

Reply:
Gotcha. Apollo Android was doing this, but I will change it.

Reply:
There are probably other places this is happening, then.

Reply:
I found and fixed one more place this was happening, with input fields.